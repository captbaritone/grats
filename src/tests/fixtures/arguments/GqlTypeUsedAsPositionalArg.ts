/** @gqlInput */
type Greeting = {
  name: string;
  salutation: string;
};

/** @gqlType */
export default class SomeType {
  /** @gqlField */
  hello(greeting: Greeting): string {
    return `${greeting.salutation} ${greeting.name}!`;
  }
}
