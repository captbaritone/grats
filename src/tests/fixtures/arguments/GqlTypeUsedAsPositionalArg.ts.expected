-----------------
INPUT
----------------- 
/** @gqlInput */
type Greeting = {
  name: string;
  salutation: string;
};

/** @gqlType */
export default class SomeType {
  /** @gqlField */
  hello(greeting: Greeting): string {
    return `${greeting.salutation} ${greeting.name}!`;
  }
}

-----------------
OUTPUT
-----------------
src/tests/fixtures/arguments/GqlTypeUsedAsPositionalArg.ts:10:19 - error: Unexpected GraphQL type used as resolver parameter. Resolver input arguments must be specified as a single `args` object literal: `args: {argName: ArgType}`.

10   hello(greeting: Greeting): string {
                     ~~~~~~~~
